00-basics/002-javascript/freeCodeCamp_03_18.js

```diff
@@ -0,0 +1,331 @@
+        return "Medium";
+    } else if (num < 20) {
+        return "Large";
+    } else if (num >= 20){
+        return "Huge";
+    }
+    return "Change Me";
+}console.log(testSize(7));
+
+let names = ["Hole-in-one!", "Eagle", "Birdie", "Par", "Bogey", "Double Bogey", "Go Home!"];
+function golfScore(par, strokes) {
+    if (strokes === 1) {
+        return names [0];
+    } else if (strokes <= par -2) {
+        return names [1];
+    } else if (strokes === par) {
+        return names [3];
+    } else if (strokes === par + 1) {
+        return names [4];
+    } else if (strokes === par + 2) {
+        return names [5];
+    } else if (strokes >= par +3) {
+        return names [6];
+    }
+    return "Change Me";
+}
+console.log(golfScore(5,4));
+
+
+function caseInSwitch(val) {
+    let answer = "";
+    switch(val) {
+        case 1:
+            return "alpha";
+            break;
+        case 2:
+            return "beta";
+            break;
+        case 3:
+            return "gamma";
+            break;
+        case 4:
+            return "delta";
+            break;
+     //Why Unreachable code? all break (181, 184, 187, 190 )
+    }
+    return answer;
+}
+console.log(caseInSwitch(1));
+
+function switchOfStuff(val) {
+    let answer = "";
+    switch (val) {
+        case "a":
+            answer = "apple";
+            break;
+        case "b":
+            answer = "bird";
+            break;
+        case "c":
+            answer = "cat";
+            break;
+        default:
+            answer = "stuff";
+    }
+   return answer;
+}
+console.log(switchOfStuff(1));
+
+function sequentialSize(val) {
+    let answer = "";
+    switch (val) {
+        case 1:
+        case 2:
+        case 3:
+            answer = "Low";
+            break;
+        case 4:
+        case 5:
+        case 6:
+            answer = "Mid";
+            break;
+        case 7:
+        case 8:
+        case 9:
+            answer = "High";
+            break;
+    }
+    return answer;
+}
+console.log(sequentialSize(5));
+
+function chainToSwitch(val) {
+    let answer = "";
+    switch (val) {
+        case "bob":
+        answer = "Marley";
+        break;
+        case 42:
+            answer = "The Answer";
+            break;
+        case 1:
+            answer = "There is no #1";
+            break;
+        case 99:
+            answer = "Missed me by this much!"
+            break;
+        case 7:
+            answer = "Ate Nine";
+            break;
+    }
+    return answer;
+}
+console.log(chainToSwitch(7));
+
+function isLess(a, b) {
+    return (a < b);
+}
+console.log(isLess(10, 15));
+
+function abTest(a, b) {
+    if (a < 0 || b < 0) {
+        return undefined;
+    }
+    return Math.round(Math.pow(Math.sqrt(a) + Math.sqrt(b), 2));
+}
+console.log(abTest(2, 2));
+
+let count = 0;
+function cc(card) {
+    switch (card) {
+        case 2:
+        case 3:
+        case 4:
+        case 5:
+        case 6:
```

Comment:
yup, you can see fall-through in effect here. cases 2-6 will all execute code in `case 6` block.